### ISSUE TITLE
[CLOSED] Vanish bug (unstealthed)

### ISSUE THREAD

**Issue by netherspite** [netherspite/cc-buglist#2324]
<a href="https://github.com/Aprics"><img src="https://avatars1.githubusercontent.com/u/13718317?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Aprics](https://github.com/Aprics)**    
_Saturday Feb 13, 2016 at 23:31 GMT_  
_Originally opened as https://github.com/ccshiro/cc-buglist/issues/2324_

----

If I cast a spell with a travel time onto a rogue that vanishes right before the spell hits him, he doesn't get hit by he spell but also doesnt really vanish.
By that I mean he has the vansish and the spell buff on him but I can still see him and cast more spells as if he was unstealthed.

This only seemed to happen on long distances with both instant and cast time spells.
It happend about 80% of the time he was standing still and not moving.
He sometimes turned invisible later at a random point.
The chance for it to happen was way lower while the rogue was moving.
Also if the effect occured and I could see the stealth buff on him, the time it took for him actually going invisible was way slower compared to standing still.

I guess the rogue should be unstealthed if he vanishes the spell too late but should lose his vanish and stealth buff.
At the moment he still has the chance too flee if he moves away fast enough without getting another cast on him.

tl:dr 
rogue vanishes too late 
cast doesnt effect him
he stays in target
has vanish and stealth as buff
can go invis by moving

**Comment by netherspite** [comment]
<a href="https://github.com/DarkenRal"><img src="https://avatars1.githubusercontent.com/u/16642872?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [DarkenRal](https://github.com/DarkenRal)**  
_Sunday Feb 14, 2016 at 05:25 GMT_

----

> I guess the rogue should be unstealthed if he vanishes the spell too late but should lose his vanish and stealth buff.

Any spell with travel time can't hit on vanish even if it was used right before it "hits".

**Comment by netherspite** [comment]
<a href="https://github.com/Aprics"><img src="https://avatars1.githubusercontent.com/u/13718317?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Aprics](https://github.com/Aprics)**  
_Sunday Feb 14, 2016 at 13:06 GMT_

----

@DarkenRal 
It was differently scrippted before though.
Something about half of the travel time.(they corrected it in the video)
So maybe it is bugged because of older scripts still working against newer ones.

**Comment by netherspite** [comment]
<a href="https://github.com/Horrorfever"><img src="https://avatars3.githubusercontent.com/u/17014772?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Horrorfever](https://github.com/Horrorfever)**  
_Sunday Feb 14, 2016 at 20:31 GMT_

----

You are literally descriping vanishing things like blind poly and anything for that matter.

Works as intended.

If you are not careful you can lose your vanish but get the spell on you anyway.

**Comment by netherspite** [comment]
<a href="https://github.com/DarkenRal"><img src="https://avatars1.githubusercontent.com/u/16642872?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [DarkenRal](https://github.com/DarkenRal)**  
_Sunday Feb 14, 2016 at 21:45 GMT_

----

No, he's not. Read carefuly:

> If I cast a spell **with a travel time** onto a rogue that vanishes right before the spell hits him, he doesn't get hit by he spell but also doesnt really vanish.

**Comment by netherspite** [comment]
<a href="https://github.com/ccnim"><img src="https://avatars3.githubusercontent.com/u/10582080?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ccnim](https://github.com/ccnim)**  
_Thursday Mar 03, 2016 at 02:36 GMT_

----

The cause for this is solely dependent on range.

**Comment by netherspite** [comment]
<a href="https://github.com/ccnim"><img src="https://avatars3.githubusercontent.com/u/10582080?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ccnim](https://github.com/ccnim)**  
_Monday Mar 07, 2016 at 11:47 GMT_

----

#1165 duplicate?
